aws/waf_helpers.go

```diff
@@ -0,0 +1,108 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func wafSizeConstraintSetSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": &schema.Schema{
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"size_constraints": &schema.Schema{
```

Comment:
Somehow I didn't notice this in my previous code review, but we tend to use singular names for TypeList/TypeSet fields with non-primitive nested types. This is because such fields are commonly represented as

```hcl
size_constraint {
  ...
}
size_constraint {
  ...
}
size_constraint {
  ...
}
```
unlike TypeList/TypeSet with primitive (e.g. TypeString) field:

```hcl
availability_zones = ["us-east-1a", "us-east-1b"]
```

However I'm 👌 with keeping it as is for this initial implementation. We can deprecate later - which we'll need to do anyway for some other WAF resources.